### PR TITLE
VitePress

### LINKED ISSUE #5: VitePress
im going to deploy a simple VitePress instance with two pages according to https://www.codementor.io/@burhanuddinahmed/deploying-vitepress-to-github-pages-1ft7mbf6lc
and then create two pages:
A landing page containing information a fake software tool from your imagination.
An installation guide for the above mentioned tool.

### PR DESCRIPTION
**Pull request by SaraLitman**
My Task was to create a simple site wich contains 2 pages:
1. A landing page containing information a fake software tool 
2. An installation guide for the above mentioned tool.

Closes #5

### CODE REVIEW

**Comment by bdsoha** [comment]
Please add a description and move **your** into a sub-directory.

**Comment by SaraLitman** [comment]
done

**Comment by bdsoha** [comment]
Where is the `vitepress` code?

**Comment by bdsoha** [comment]
It is important to follow conventions when coding, it makes the overall repo look as if it was developed by a single person, when in fact there were multiple contributors.

- Stick to `kabob-case` for file/folder names.
- Stick to `snake_case` for python files.

**Comment by SaraLitman** [comment]
Which folders/files should I rename?

**Comment by bdsoha** [comment]
`VitePress` ➡ `vitepress`.

> **Note** You will notices that the changes will have no effect because Window's filesystem is *case-insensitive*.

You will need to be creative in order to apply the changes to `git`.

**Comment by SaraLitman** [comment]
By running this image I'm getting this error:
/bin/sh: [npm,: not found
may I get Help?...

File: VitePress/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+COPY . .
+
+RUN npm install
+#RUN npm run build vitepress
+
+
+
+EXPOSE 5173
+
+CMD ["npm", "run",  "config.js\index.md"]
```
Comment (bdsoha) [inline VitePress/Dockerfile:14]: The docker image is running a `Linux` filesystem which doesn't allow backslashes.

Comment (SaraLitman) [reply VitePress/Dockerfile:14]: Now I get this error:
npm ERR! Missing script: "docs/index.md"
Could it be that my entry point is just a command and not a file?

Comment (bdsoha) [reply VitePress/Dockerfile:14]: There is a lot going on.
You have 12 files in your commit-tree.
Maybe by mistake you copied too many items.

Comment (SaraLitman) [reply VitePress/Dockerfile:14]: the right directory is vitepress, i'm trying to delete the directory named "VitePress"

**Comment by bdsoha** [comment]
You need to be using `git` from your local machine with a local `clone`. Uploading files can turn into a pain, imagine you have 20 files, or 50, or 100???

> **Warning**: It is not best practice to use the web-ui to develop projects.

**Comment by SaraLitman** [comment]
Ok, but now may I get help with bthe error I that I got?

**Comment by bdsoha** [comment]
> Ok, but now may I get help with bthe error I that I got?

It depends what the error was?

**Comment by SaraLitman** [comment]
npm ERR! Missing script: "docs/index.md"

**Comment by bdsoha** [comment]
You can't invoke `npm` on an `.md` file. You need to be using the `npm run vitepress ...` commands.

**Comment by SaraLitman** [comment]
now its anoother error...:
node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module '/app/npm run vitepress'
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
    at node:internal/main/run_main_module:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
Does this mean that something in the way I based the image from docker is wrong and therefore it doesn't know the Vitepress module?

**Comment by bdsoha** [comment]
Can you `pull` the changes and have a look.

**Comment by SaraLitman** [comment]
I did pull,
now I got this error:
 => ERROR [3/3] RUN npm install                                                                                    0.7s
------
 > [3/3] RUN npm install:
#6 0.631 npm ERR! Tracker "idealTree" already exists
#6 0.632
#6 0.633 npm ERR! A complete log of this run can be found in:
#6 0.633 npm ERR!     /root/.npm/_logs/2023-03-19T11_43_45_224Z-debug-0.log
------
executor failed running [/bin/sh -c npm install]: exit code: 1

In Docker I already overcame such an error, but I no longer remember in what form

**Comment by bdsoha** [comment]
The root directory was being overwritte. Two ways to solve:
- Set `WORKDIR /app`
- Copy to absolute path

**Comment by SaraLitman** [comment]
> docs:preview
> vitepress preview config.js


  vitepress v1.0.0-alpha.60

failed to start server. error:
Error: ENOENT: no such file or directory, open '/app/config.js/.vitepress/dist/404.html'
    at Object.openSync (node:fs:585:3)
    at Object.readFileSync (node:fs:453:35)
    at serve (file:///app/node_modules/vitepress/dist/node/serve-23671816.js:55957:32)

**Comment by bdsoha** [comment]
Try now, I just ran it in a codespace. Works great.

**Comment by SaraLitman** [comment]
I'm geeting this error again,
Could it be because npm is not running on my computer? (allegedly something in the node installation)

**Comment by bdsoha** [comment]
> I'm geeting this error again, Could it be because npm is not running on my computer? (allegedly something in the node installation)

Did you rebuild the docker image?
You should not be running `node` on your machine.

**Comment by SaraLitman** [comment]
yes, I did rebuild

**Comment by bdsoha** [comment]
What error are you seeing?

**Comment by SaraLitman** [comment]
Now I tried to rebuild again and it fail with the error "=> ERROR [5/5] RUN npm run build "
I did pull before.

**Comment by bdsoha** [comment]
> Now I tried to rebuild again and it fail with the error "=> ERROR [5/5] RUN npm run build " I did pull before.

`build` doesn't exist, `docs:build` does.
I made the change.

**Comment by SaraLitman** [comment]
i got this , it seems OK.
I can't check in the browser because I ran into a firewall

> docs:preview
> vitepress preview config.js


  vitepress v1.0.0-alpha.60

Built site served at http://localhost:4173/

**Comment by bdsoha** [comment]
Don't forget ports. Replace `{YOUR_IMAGE_NAME}` with your image's name.

```sh
docker run -it --rm -p 4137:4137 {YOUR_IMAGE_NAME}
```

**Comment by SaraLitman** [comment]
Still running into the firewall...

**Comment by bdsoha** [comment]
> Still running into the firewall...

There is no concept of *firewall* when it comes to running docker locally.
It is either port forwarding *(i.e. `-p 4137:4137`)* or that within the docker container, the server is not listening on all hosts *(i.e. `0.0.0.0`)*.

**Comment by bdsoha** [comment]
Let me test it in a codespace...

**Comment by SaraLitman** [comment]
still running into problem,
Do I need to test the run at a different address than http://localhost:4173/? I also tried http://localhost:5173/

**Comment by bdsoha** [comment]
I was able to run the container.
Since we are press with time for this assignment, I will merge it as is.

**Comment by bdsoha** [comment]
🎉 Congrats on your first merged PR.

You will soon be added as a collaborator to this repo.
In the meantime, please prepare a short presentation (in English) describing:

The task in general.
Things you learned along the way.
Tips for other team-members for future tasks.

**Comment by bdsoha** [comment]
You have been sent an email invitation to the [SkyVar Organization](https://github.com/skyvar).
Please open your cooperate email and finish the registration process.

**Pull request merged**